pkg/services/ngalert/schedule/registry.go

```diff
@@ -109,10 +111,23 @@ func (a *alertRuleInfo) eval(t time.Time, version int64) (bool, *evaluation) {
 	}
 }
 
-// update signals the rule evaluation routine to update the internal state. Does nothing if the loop is stopped
-func (a *alertRuleInfo) update() bool {
+// update signals the rule evaluation routine to update the internal state. Does nothing if the loop is stopped.
+func (a *alertRuleInfo) update(version ruleVersion) bool {
+	// check if the channel is not empty.
+	msg := version
+	select {
+	case v := <-a.updateCh:
```

Comment:
I don't think this code is required. I think it's also possible that `update` can starve the rule routine from receiving `updateCh` messages, depending on how frequent `updateCh` is called and the timing of when an evaluation is started and has completed.

Reply:
I think it will help us be consistent on how we treat the rule evaluation loop. In the eval channel, we drop ticks as well.